### ISSUE TITLE
Chapter 3: list average function causes no errors

### ISSUE THREAD

**Issue by nekketsuuu** [ucsd-progsys/liquidhaskell-tutorial#38]
Hello. I'm reading chapter 3, and I may find a problem with it.

At [list average problem](https://github.com/ucsd-progsys/liquidhaskell-tutorial/blob/master/src/03-basic.lhs#L276), it says `avg` caused a LiquidHaskell error. But it doesn't.

```
src$ liquid 03-basic.lhs
(omit)
**** RESULT: UNSAFE ************************************************************

 03-basic.lhs:70:1-4: Error: Liquid Type Mismatch
 70 | one' = 1 :: Int
      ^^^^
 03-basic.lhs:393:23-47: Error: Liquid Type Mismatch
 393 | lAssert False _ = die "yikes, assertion fails!"
                             ^^^^^^^^^^^^^^^^^^^^^^^^^
```

However it fails successfully when copy and paste `avg` from chapter 3.

```
$ liquid avg.hs
(omit)
**** RESULT: UNSAFE ************************************************************

 avg.hs:15:10-23: Error: Liquid Type Mismatch
 15 | avg xs = divide total n
               ^^^^^^^^^^^^^^
```

Why does this difference occur? All logs is in [this gist](https://gist.github.com/nekketsuuu/f52be2422fb16b604bebbae8eaea01b5). I use LiquidHaskell-6.0.0 (precisely, I install LiquidHaskell from source at the point of [May 4, 2016](https://github.com/ucsd-progsys/liquidhaskell/tree/99e8c7cad8a2fa6528d7db6a0ef5865af45dd3be)) and GHC 7.10.3.

**Comment by ranjitjhala** [comment]
Thanks for pointing this out! Due to some recent changes, it seems that the `one'` binder (which has an incorrect signature -- and is correctly flagged) ends up *suppressing* the later error in `avg`. I have fixed the tutorial so that the `one'` binder is now *not* a top-level one. Consequently you should see the error.

Many thanks again!

/cc @nikivazou

**Comment by ranjitjhala** [comment]
Fixed by 94b9318453091e4d86e20403fdd808004e9f4dc1

**Comment by nekketsuuu** [comment]
Thank you!
